docs/proposals/excluding-drives.md

```diff
@@ -0,0 +1,108 @@
+# Proposal: Specific drives excluding
+
+Last updated: 07.02.2022
+
+
+## Abstract
+
+CSI Baremetal needs to have API for excluding specific drives from the common storage pool. 
+Excluded drives won't be used for allocating new Volumes or LVGs.
+
+## Background
+
+CSI customers might not want to use some disks for whatever reason:
+- for another application not managed by CSI
+- as a workaround for any disk detection issues in CSI
+
+So in these cases CSI haven't to allocate Volumes for some drives.
+
+## Proposal
+
+### User API
+
+The rules to include/exclude drives based on:
+- Size
+- Type (HDD, SSD, ...)
+- Serial Number
+- PID
+- VID
+```yaml
+drive_rule:
+  size:
+    less_then: 1Gi
+    more_then: 100Mi
+  # include drives with the following parameters only
+  in:
```

Comment:
Do we really need to have "in" list? Maybe it would be better to have only "not_in" drives list? So, all the drives are acceptable by default, and drives from "not_in" list has to be excluded from CSI usage.

Reply:
@mitsaa @rkiselenko it was my request to support include/exclude filtering options. We don't know all the use cases, but other CSI drivers work with **include** option only.

Reply:
From my prospective, it is not clear what to do if a drive satisfies conditions from both filters "include" and "non-include".
If most of CSI drivers use one "include" option, it makes sense to follow common practice, I believe. Folks, can you please your views on this?

Reply:
Auto disk defining is the feature of our CSI, isn't it? :) In the case described below comment, we can have include/exclude rules for different option types (include all HDD, but exclude some drives via SN or PID). 
I agree that the rules struct here can bring some satisfies conditions, which will be user responsibility only. Could you advise on how we can clarify `drive_rule` to avoid it?

Reply:
If our CSI will be used on unrestricted HW configurations, we may have more issues than benefits with accepting to usage disks in auto mode. There are two ways here:
-  Accept all the disks excluding deprecated in "not_included" list/ConfigMap;
- Accept only disks from "included"  list/ConfigMap.
My tip is to have only **one** list/template of allowed **or** deprecated drives.

Reply:
I think user will have to choose which strategy to follow:
1. Include rules
2. Exclude rules
Mixed mode should not be allowed

Reply:
@mishoyama, two strategies looks cool. I am agree with that. 
BTW, is that possible to use something like regexp filters to set templates? I'm not sure that it would be useful to set definite VID/PID/SN-es for every node.

Reply:
@mitsaa I suggest to start with the simple implementation until we get feedback from users with their use cases.

Reply:
Thank you! `driveRule` struct was updated. Please, check

Reply:
OK, let's get feedback from users.